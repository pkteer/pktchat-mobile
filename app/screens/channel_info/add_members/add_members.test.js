// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import AddMembers from './add_members';

jest.mock('@utils/theme', () => {
    const original = jest.requireActual('../../../utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('ChannelInfo -> Add Members', () => {
    const baseProps = {
        canManageUsers: true,
        groupConstrained: false,
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <AddMembers
                {...baseProps}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should render null if cannot manage members', () => {
        const wrapper = shallowWithIntl(
            <AddMembers
                {...baseProps}
                canManageUsers={false}
            />,
        );
        expect(wrapper.getElement()).toBeNull();
    });

    test('should render null if channel is constrained to groups', () => {
        const wrapper = shallowWithIntl(
            <AddMembers
                {...baseProps}
                groupConstrained={true}
            />,
        );
        expect(wrapper.getElement()).toBeNull();
    });
});
