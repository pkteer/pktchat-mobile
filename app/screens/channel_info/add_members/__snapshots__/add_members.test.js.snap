// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo -> Add Members should match snapshot 1`] = `
<React.Fragment>
  <Separator
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
  <channelInfoRow
    action={[Function]}
    defaultMessage="Add Members"
    icon="account-plus-outline"
    rightArrow={true}
    shouldRender={true}
    textId="channel_header.addMembers"
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
    togglable={false}
  />
</React.Fragment>
`;
